user/super/com/google/gwt/emul/java/lang/CaseMapper.java

```diff
@@ -23,7 +23,23 @@ public static char charToLowerCase(char c) {
   }
 
   public static char charToUpperCase(char c) {
-    return String.valueOf(c).toUpperCase().charAt(0);
+    String upper = String.valueOf(c).toUpperCase();
+    return hasExtraCodePoints(upper) ? c : upper.charAt(0);
+  }
+
+  public static int intToLowerCase(int codePoint) {
+    return String.NativeString.fromCodePoint(codePoint).toLowerCase().codePointAt(0);
+  }
+
+  public static int intToUpperCase(int codePoint) {
+    String upper = String.NativeString.fromCodePoint(codePoint).toUpperCase();
+    return hasExtraCodePoints(upper) ? codePoint : upper.codePointAt(0);
+  }
+
+  // If String.toUpperCase produces more than 1 codepoint, Character.toUpperCase should
+  // act either as identity or title-case conversion (not supported in GWT).
```

Comment:
Maybe add a failing test and mark it ignored so we can see about restoring it when it works?